tests/testthat/testEdgeThresh.R

```diff
@@ -0,0 +1,251 @@
+nLargest <- max(igraph::components(EGG)$csize)
+permutedGraph <- .permuteGraph(EGG)
+nVertices <- igraph::vcount(EGG)
+nEdges <- igraph::ecount(EGG)
+# ==== END SETUP AND PREPARE ===================================================
+
+
+test_that("edgeThresh parameter errors are correctly handled", {
+    # Try each parameter missing in turn (only EGG has no defaults)
+    expect_error(edgeThresh())
+
+    # Try parameters out of expected bounds
+    expect_error(edgeThresh(myEGG, Lmax =  1))
+    expect_error(edgeThresh(myEGG, Lmax =  0))
+    expect_error(edgeThresh(myEGG, Lmax = -1))
+    expect_error(edgeThresh(myEGG, Lmax = nLargst - 1))
+
+    expect_error(edgeThresh(myEGG, Lmax = 20, N =  0))
+    expect_error(edgeThresh(myEGG, Lmax = 20, N = -1))
+
+    expect_error(edgeThresh(myEGG, Lmax = 20, Q =  0))
+    expect_error(edgeThresh(myEGG, Lmax = 20, Q = -1))
+
+
+    # Try NULL and length-zero parameters: be sure none lead to an erroneous
+    #    one-time execution of any loop: NULL, logical, character
+    expect_error(edgeThresh(EGG = NULL,  Lmax = 20))
+    expect_error(edgeThresh(EGG = myEGG, Lmax = 20, N = NULL))
+    expect_error(edgeThresh(EGG = myEGG, Lmax = 20, Q = NULL))
+    expect_error(edgeThresh(EGG = myEGG, Lmax = NULL))
+
+    expect_error(edgeThresh(EGG = TRUE))
+    expect_error(edgeThresh(EGG = myEGG, Lmax = 20, N = TRUE))
+    expect_error(edgeThresh(EGG = myEGG, Lmax = 20, Q = TRUE))
+    expect_error(edgeThresh(EGG = myEGG, Lmax = TRUE))
+
+    expect_error(edgeThresh(EGG = "myEGG", Lmax =  20))
+    expect_error(edgeThresh(EGG =  myEGG , Lmax = "20"))
+    expect_error(edgeThresh(EGG =  myEGG , Lmax =  20, Q = "100"))
+    expect_error(edgeThresh(EGG =  myEGG , Lmax =  20, N = "100"))
+
+})
+
+# ((Does .permuteGraph need parameter testing? or is it assumed that
+# they are valid since the calling function has already checked them?))
+test_that(".permuteGraph parameter errors are correctly handled") {
+
+    expect_error(.permuteGraph())
+    expect_error(.permuteGraph(graph = myEGG))
+    expect_error(.permuteGraph(Q = 100))
+
+    expect_error(.permuteGraph(graph = NULL,  Q = 100))
+    expect_error(.permuteGraph(graph = myEGG, Q = NULL))
+
+    expect_error(.permuteGraph(graph = TRUE,  Q = 100))
+    expect_error(.permuteGraph(graph = myEGG, Q = TRUE))
+
+    expect_error(.permuteGraph(graph ="myEGG",Q =  100 ))
+    expect_error(.permuteGraph(graph = myEGG, Q = "100"))
+
+    expect_error(.permuteGraph(graph = myEGG, Q =  0))
+    expect_error(.permuteGraph(graph = myEGG, Q = -1))
+
+    expect_error(.permuteGraph(graph = 10,    Q = 100))
+
+}
+# ((Should valid inputs different from rete.EGGprototype be used?))
+# ((The testing below can get quite computationally expensive...))
+test_that("a sane input to .permuteGraph gives an expected output", {
+
+    ### Test for Q = 1, 10, 20, 50, 100, 200
+    myPermutation <- .permuteGraph(myEGG, Q = 1)
+
+    # Check for self-loops and multiple edges
```

Comment:
Why are you not answering the question? First you are questioning whether the question is valid, thereby contradicting the reason of your test. Then you are reading an implication of a new requirement into my question. Just answer the simple question: what happens if the graph is not simple?